Windows/XinputDevice.cpp

```diff
@@ -344,20 +366,18 @@ void XinputDevice::ApplyButtons(int pad, const XINPUT_STATE &state) {
 }
 
 
-void XinputDevice::ApplyVibration(int pad) {
-	if (PSP_IsInited) {
-		XINPUT_VIBRATION vibration;
-		ZeroMemory(&vibration, sizeof(XINPUT_VIBRATION));
-		vibration.wLeftMotorSpeed = 0;
-		vibration.wRightMotorSpeed = 0;
-		
-		engine_ = new CWCheatEngine(gameID_);
+void XinputDevice::ApplyVibration(int pad, XINPUT_VIBRATION &vibration) {
+	if (PSP_IsInited()) {
 		if (engine_) {
-			vibration.wLeftMotorSpeed = engine_->GetLefttVibration(); // use any value between 0-65535 here
+			vibration.wLeftMotorSpeed = engine_->GetLeftVibration(); // use any value between 0-65535 here
 			vibration.wRightMotorSpeed = engine_->GetRightVibration(); // use any value between 0-65535 here
 		}
-		WARN_LOG(COMMON, "PAD TEST LEFT: %d, RIGHT: %d, PAD: %d", vibration.wLeftMotorSpeed, vibration.wRightMotorSpeed, pad);
-		XInputSetState(pad, &vibration);
+		if (prevVibration[pad].wLeftMotorSpeed != vibration.wLeftMotorSpeed || prevVibration[pad].wRightMotorSpeed != vibration.wRightMotorSpeed) {
+			PPSSPP_XInputSetState(pad, &vibration);
+			WARN_LOG(COMMON, "PAD TEST LEFT: %d / %d, RIGHT: %d / %d, PAD: %d, engine_: %d", vibration.wLeftMotorSpeed, prevVibration[pad].wLeftMotorSpeed, vibration.wRightMotorSpeed, prevVibration[pad].wRightMotorSpeed, pad, engine_);
+			prevVibration[pad] = vibration;
+		}
+		KeyMap::UpdateNativeMenuKeys();
```

Comment:
Hm?

-[Unknown]

Reply:
Sorry, a leftover when I tried random things from exiting settings, removed it already, moving things as you proposed earlier to sceCtrl now.